packages/core/src/builtin-tasks/node.ts

```diff
@@ -23,64 +25,122 @@ const TERMINATION_SIGNALS = ['SIGINT', 'SIGTERM', 'SIGQUIT']
 const BUILTIN_SCRIPTS = ['anyone_can_pay', 'omni_lock', 'simple_udt']
 const DEFAULT_BUILTIN_CONTRACT_DOWNLOAD_BASE_URL =
   'https://github.com/ckb-js/ckb-production-scripts/releases/download/scripts'
+const DEFAULT_CKB_BIN_DOWNLOAD_BASE_URL = 'https://github.com/nervosnetwork/ckb/releases/download'
+const DEFAULT_CKB_BIN_VERSION = 'v0.109.0'
+
+const startBinNode = async (version: string, port: number, genesisArgs: string[]): Promise<CKBNode> => {
+  const packageName = `ckb_${version}_${os.machine()}-${osPlatform()}`
+  const packageFileName = `${packageName}.${packageType()}`
+
+  if (!fs.existsSync(path.resolve(cachePath('ckb', 'bin'), packageName))) {
+    await download(
+      `${DEFAULT_CKB_BIN_DOWNLOAD_BASE_URL}/${version}/${packageFileName}`,
+      path.resolve(cachePath('ckb', 'zip'), `${packageFileName}`),
+    )
+
+    if (osPlatform() === 'unknown-linux-gnu') {
```

Comment:
I'm not sure if the condition is proper. I would use a condition of if `tar` or `unzip` exists.

Reply:
> I'm not sure if the condition is proper. I would use a condition of if `tar` or `unzip` exists.

I think I would better to checking whether the extract tool exists or not first. If not, I will print an error message.